### ISSUE TITLE
Encryption for cloud backup

### ISSUE THREAD

**Issue by willbasky** [encryptedcoins/encoins-frontend#135]
After #81 

how we should encrypt the backup tokens.

1. Each token is defined by two numbers v and gamma. We convert v to an 8-byte bytestring and gamma to a 32-byte bytestring.
2. we encrypt using AES-256.
3. Verify that this is our token when restoring from backup: decrypt back to v and gamma and verify that 0 <= v < 2^20. The rest of the bits for v just play the role of cheque sum, which should be zero if this is our token.
4. Accordingly, the user will need to write a 256-bit key to enable the backup. For now we can just show it in hex. Making the cid phrase will be a separate task.
5. Depending on the password, this key will be stored either in clear or encrypted form in the local cache. In case of restoring from backup it will have to be entered. That is, the password will not be related to the backup in any way - it is only used for the local cache.
